src/Functions/FunctionsExternalDictionaries.h

```diff
@@ -76,6 +76,9 @@ class FunctionDictHelper : public WithContext
         auto current_context = getContext();
         auto dict = current_context->getExternalDictionariesLoader().getDictionary(dictionary_name, current_context);
 
+        if (dict->getDatabaseOrNoDatabaseTag() == DatabaseCatalog::TEMPORARY_DATABASE || dict->getDictionaryID().getTableName().starts_with("_tmp_replace_"))
```

Comment:
For the dictionary can we check like this for temporary one? (i.e. which is not yet stored)
`(!dict->getDictionaryID().hasUUID()` ?